src/transaction.rs

```diff
@@ -92,15 +92,20 @@ pub struct Transaction {
 
     /// The number of tokens paid for processing and storage of this transaction.
     pub fee: i64,
+    /// Optional user data to be stored in the account
+    /// TODO: This will be a required field for all contract operations including a simple spend.
+    /// `instruction` will be serialized into `userdata` once Budget is its own generic contract.
+    pub userdata: Option<Vec<u8>>,
 }
 
 impl Transaction {
-    /// Create a signed transaction from the given `Instruction`.
-    fn new_from_instruction(
+    /// Create a signed transaction with userdata and an instruction
+    fn new_with_userdata_and_instruction(
         from_keypair: &Keypair,
         instruction: Instruction,
         last_id: Hash,
         fee: i64,
+        userdata: Option<Vec<u8>>,
```

Comment:
Is there a semantic difference between `None` and `vec![]`? In no, can the `Option<>` be removed?

Reply:
7 bytes of data extra per serialized transaction.

Reply:
Yep, yep, but you mentioned that your next step in #953 is to replace the `instruction` field with this, at which point there would be no use for the `None` case, right? You can save us a bunch of code churn by getting rid of that `Option` now.